src/core/src/package_managers/ZypperPackageManager.py

```diff
@@ -44,13 +44,17 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
         # Package manager exit code(s)
         self.zypper_exitcode_ok = 0
         self.zypper_exitcode_zypper_updated = 103
+        self.zypper_exitcode_zypp_locked = 7
 
         # Support to check for processes requiring restart
         self.zypper_ps = "sudo zypper ps -s"
 
         # Miscellaneous
         self.set_package_manager_setting(Constants.PKG_MGR_SETTING_IDENTITY, Constants.ZYPPER)
         self.zypper_get_process_tree_cmd = 'ps --forest -o pid,cmd -g $(ps -o sid= -p {})'
+        self.env_layer.ensure_env_var_is_set('ZYPP_LOCK_TIMEOUT', 5)
```

Comment:
Read what the original value is first (and log). Do all of this as you enter invoke_package_manager and reset to original value when you exit from the function. Setting env. var is a low cost operation so the repeats are okay. We shouldn't leave permanent configuration changes on customer machines. There is still a possibility of failure where the value is not reset, but we can address that case later.

Reply:
Done. Split into get/set and moved to invoke_package_manager